addons/web/static/tests/helpers/qunit_config.js

```diff
@@ -40,15 +40,54 @@ QUnit.config.hidepassed = (window.location.href.match(/[?&]testId=/) === null);
 
 var sortButtonAppended = false;
 
+/**
+ * Waits for the module system to end processing the JS modules, so that we can
+ * make the suite fail if some modules couldn't be loaded (e.g. because of a
+ * missing dependency).
+ *
+ * @returns {Promise<boolean>}
+ */
+async function checkModules() {
+    // do not mark the suite as successful already, as we still need to ensure
+    // that all modules have been correctly loaded
+    $('#qunit-banner').removeClass('qunit-pass');
+    const $modulesAlert = $('<div>')
+        .addClass('alert alert-info')
+        .text('Waiting for modules check...');
+    $modulesAlert.appendTo('#qunit');
+
+    // wait for the module system to end processing the JS modules
+    await odoo.__DEBUG__.didLogInfo;
+
+    const info = odoo.__DEBUG__.jsModules;
+    if (info.missing.length || info.failed.length) {
+        $('#qunit-banner').addClass('qunit-fail');
+        $modulesAlert.toggleClass('alert-info alert-danger');
+        const failingModules = info.missing.concat(info.failed);
+        const error = `Some modules couldn't be started: ${failingModules.join(', ')}.`;
+        $modulesAlert.text(error);
+        console.error(error);
+        return false;
+    } else {
+        $modulesAlert.toggleClass('alert-info alert-success');
+        $modulesAlert.text('All modules have been correctly loaded.');
+        $('#qunit-banner').addClass('qunit-pass');
+        return true;
+    }
+}
+
 /**
  * This is the way the testing framework knows that tests passed or failed. It
  * only look in the phantomJS console and check if there is a ok or an error.
  *
  * Someday, we should devise a safer strategy...
  */
-QUnit.done(function(result) {
+QUnit.done(async function (result) {
     if (!result.failed) {
-        console.log('test successful');
+        const allModulesLoaded = await checkModules();
```

Comment:
I would check modules even if the a test failed so both errors can be reported as soon as possible instead of needing another pass.
I think a failed module could even explain why the test in question failed.

Reply:
I can do something like this

> 2020-04-23 10:08:59,535 31358 ERROR (...): FAIL: IMSupportSuite.test_im_support_js
Traceback (most recent call last):
  File "/home/odoo/Coding/odoo/addons/im_support/tests/test_js.py", line 11, in test_im_support_js
    self.phantom_js('/im_support/tests?mod=web&failfast', "", "", login='admin', timeout=180)
  File "/home/odoo/Coding/odoo/odoo/tests/common.py", line 1226, in browser_js
    self.fail('%s\n%s' % (message, error))
AssertionError: Some js test failed
1 / 2 tests failed.
"QUnit test failed: "im_support > systray > messaging menu displays the Support channel" [message: "Expected 2 assertions, but 1 were run", expected: "undefined"]
Some modules couldn't be started: web.AbstractView2